### ISSUE TITLE
spin build with watch config but no command config fails

### ISSUE THREAD

**Issue by vdice** [fermyon/spin#1443]
If an app has a component with a `[component.build]` section containing `watch` configuration but no `command` configuration, `spin build` (using version `spin 1.2.0-pre0 (5bcc14b 2023-05-01)`) will error out with:

```
$ spin build
Error: missing field `command` at line 18 column 1
```

Encountered the error when attempting to build a new app created from the current [http-php template](https://github.com/fermyon/spin/tree/main/templates/http-php).

I've created this issue assuming this is a bug; meaning, perhaps we do wish to support build sections that don't necessarily contain `command` config (i.e. may only contain `watch` config)

Or, perhaps this isn't something we support (i.e. is not a bug) and we should update the http-php template (appears to be the only app in source with this configuration).

**Comment by itowlson** [comment]
Agree we should handle this case gracefully, but yeah, I would update the PHP template too: the files in its `build.watch` are never built, and are included in the application as assets rather than built to Wasm.  It's possible it relates to an earlier iteration of `spin watch` where source and artifact watches were modelled a bit differently.  @calebschoepp is there a reason to keep the PHP `build.watch` or is it okay for me to biff it?

(and thanks for checking the other templates @vdice!)

**Comment by calebschoepp** [comment]
Let's biff it. No reason to keep it and I think I just didn't understand how the PHP app was working.

**Comment by itowlson** [comment]
Apparently it's syntactically invalid to have a `build` section without a `command` so the error is actually a serialiser error saying the `spin.toml` file is bad.  Sorry, @vdice, I completely misread your report (even though it says `spin build` RIGHT THERE IN THE TITLE).  There's no action for `spin watch` here; it's purely a PHP template bug, addressed in #1445.

**Comment by itowlson** [comment]
Closing in favour of #1445 because that already has the fix PR associated with it.
